docs/internals/optimizer.rst

```diff
@@ -1096,6 +1097,48 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and ``mstore`` statements if conditions are met for said removal.
+In case of an ``sstore``, if all outgoing code paths lead to a revert or to another ``sstore`` for which the optimizer
+can tell that it will overwrite the first store, the statement will be removed. For example, the following code
+
+.. code-block:: yul
+
+    {
+        let c := calldataload(0)
+        sstore(c, 1)
+        if c {
+            sstore(c, 2)
+        }
+        sstore(c, 3)
+    }
+
+will produce the code below after the unused store eliminator step is run
+
+.. code-block:: yul
+
+    {
+        {
+            let c := calldataload(0)
+            let _2 := 1
+            if c { let _3 := 2 }
+            sstore(c, 3)
+        }
+    }
+
+For ``mstore``, things are generally simpler, at least in the outermost yul block as all ``mstore``
+statements will be removed if they are never read from in all code paths. At function analysis level however,
```

Comment:
It's best to break lines at the end of the sentence. RST will join lines anyway but this means you do not have to rewrap the whole paragraph when you modify just one sentence. Diffs are also cleaner:
```suggestion
statements will be removed if they are never read from in all code paths.
At function analysis level however, the approach is similar to ``sstore``, as we don't know
```

Reply:
> It's best to break lines at the end of the sentence.
Sure, I'll re-work it, but I'm assuming this doesn't apply to longer sentences.

Reply:
For longer sentences I'd just break them in the middle (around 100 chars). The whole sentence is then still a complete unit that can be modified or removed without having to rewrap stuff around it.